src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,394 @@
+        return this.generateUserAssignments(args);
+      })
+      .then(assignments => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/tasks`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none'
+          },
+          responseType: 'json',
+          data: {
+            planId: args.options.planId,
+            bucketId: args.options.bucketId,
+            title: args.options.title,
+            startDateTime: args.options.startDateTime,
+            dueDateTime: args.options.dueDateTime,
+            percentComplete: args.options.percentComplete,
+            assignments: assignments,
+            orderHint: args.options.orderHint
+          }
+        };
+
+        return request.post(requestOptions) as PlannerTask;
+      })
+      .then(newTask => { 
+        return this.updateTaskDetails(args.options, newTask);  
+      })
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getTaskDetailsEtag(taskId: string): Promise<string> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/tasks/${encodeURIComponent(taskId)}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => {
+        const etag: string | undefined = response ? response['@odata.etag'] : undefined;
+
+        if (!etag) {
+          return Promise.reject(`Error fetching task details`);
+        }
+
+        return Promise.resolve(etag);
+      });
+  }
+
+  private updateTaskDetails(options: Options, newTask: PlannerTask): Promise<PlannerTask & PlannerTaskDetails> {
+    const taskId = newTask.id as string;
+
+    if (options.description) {
+      return this.getTaskDetailsEtag(taskId)
+        .then(etag => {
+          const requestOptionsTaskDetails: any = {
+            url: `${this.resource}/v1.0/planner/tasks/${taskId}/details`,
+            headers: {
+              'accept': 'application/json;odata.metadata=none',
+              'If-Match': etag,
+              'Prefer': 'return=representation'
+            },
+            responseType: 'json',
+            data: {
+              description: options.description
+            }
+          };
+  
+          return request.patch(requestOptionsTaskDetails);              
+        })
+        .then(taskDetails => {
+          return { ...newTask, ...taskDetails as PlannerTaskDetails };
+        });
+    } 
+    else {
+      return Promise.resolve(newTask);
+    }  
+  }
+
+  private generateUserAssignments(args: CommandArgs): Promise<{ [userId: string]: { [property: string]: string }; }> {
+    const assignments: { [userId: string]: { [property: string]: string } } = {};
+
+    if (args.options.assignedToUserNames || args.options.assignedToUserIds) {
```

Comment:
To make the code more readable, let's invert the clause to `if (!args.options.assignedToUserIds && !args.options.assignedToUserNames)` and use an early return.